buildenv/jenkins/jobs/builds/Build-Test-Any-Platform

```diff
@@ -0,0 +1,73 @@
+/*******************************************************************************
+ * Copyright (c) 2018, 2018 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+ARGS = ['SDK_VERSION', 'PLATFORM']
+
+node('master') {
+    timestamps {
+        try{
+            retry(5){
+                checkout scm
+            }
+
+            variableFile = load 'buildenv/jenkins/common/variables-functions'
+
+            if (!params.UPSTREAM_JOB_NAME && !params.UPSTREAM_JOB_NUMBER) {
+                variableFile.set_job_variables('build')
+                buildFile = load 'buildenv/jenkins/common/build'
+            }
+
+            if (run_tests()) {
+                variableFile.set_job_variables('test')
+            }
+        } finally {
+            try{
+                cleanWs()
+            } catch(e) {
+                variableFile.printStackTrace(e)
+            }
+        }
+    }
+}
+
+node("${NODE}") {
+    if (!run_tests()) {
```

Comment:
We don't have a `try/finally` with `cleanWs()` here, presumably because both `build_all()` and `test_all()` have that functionality. Is it better to move that functionality here and not duplicate them in the `build` and `test` files? It might also allow us to remove `build_pr()` since that function is designed to not cleanup after a PR compile. That way we could have something more like, `if compile; build_all; if test; test_all; done`

Reply:
There are 130 Jenkins files still relying on build and test scripts. timeout, timestamps, cleanWs() should be done here but at this point it only duplicates an existing functionality. The refactoring should take place after we replace all the platform specific Jenkins files with Pipeline-Buld-Test-Any-Platform, Buld-Test-Any-Platform scripts.